src/System.Text.Json/src/System/Text/Json/JsonHelpers.Date.cs

```diff
@@ -226,14 +226,13 @@ private static bool TryParseDateTimeOffset(ReadOnlySpan<byte> source, out DateTi
             // Parse fraction
             {
                 int numDigitsRead = 0;
+                int fractionEnd = Math.Min(sourceIndex + JsonConstants.DateTimeParseNumFractionDigits, source.Length);
 
-                while (sourceIndex < source.Length && IsDigit(curByte = source[sourceIndex]))
+                while (sourceIndex < fractionEnd && IsDigit(curByte = source[sourceIndex]))
                 {
-                    int prevFractionTimesTen = fraction * 10;
-
-                    if ((prevFractionTimesTen + (int)(curByte - (uint)'0') <= JsonConstants.MaxDateTimeFraction) && (numDigitsRead < JsonConstants.DateTimeNumFractionDigits))
+                    if (numDigitsRead < JsonConstants.DateTimeNumFractionDigits)
                     {
-                        fraction = prevFractionTimesTen + (int)(curByte - (uint)'0');
+                        fraction = (fraction * 10) + (int)(curByte - (uint)'0');
```

Comment:
In the PR, you mentioned "Also, if we don't enforce a cap, we could overflow at 10 digits." This is a good observation! It should also exist as a code comment here so that somebody's not tempted to change the value later. :)